src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/service/SubsystemService.java

```diff
@@ -61,4 +66,19 @@ private void saveSecurityServerClientName(Subsystem subsystem) {
         securityServerClientNameRepository.save(ssClientName);
     }
 
+    public void deleteSubsystem(ClientId subsystemClientId) {
+        var subsystem = subsystemRepository.findOneBy(subsystemClientId)
+                .getOrElseThrow(() -> new NotFoundException(ErrorMessage.SUBSYSTEM_NOT_FOUND));
+
+        if (isRegistered(subsystem)) {
+            throw new ValidationFailureException(ErrorMessage.SUBSYSTEM_REGISTERED_AND_CANNOT_BE_DELETED);
+        }
+
+        subsystemRepository.delete(subsystem);
+    }
+
+    private boolean isRegistered(Subsystem subsystem) {
+        return !subsystem.getServerClients().isEmpty();
```

Comment:
Can `subsystem.getServerClients()` never be null?
If it can then would be good a idea to replace with: 
```suggestion
        return !CollectionUtils.isEmpty(subsystem.getServerClients())
```